File: backend/src/main/java/com/servis/backend/entity/Product.java
package com.servis.backend.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "products")
public class Product {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, length = 200)
    private String name;

    @Column(nullable = false, unique = true, length = 220)
    private String slug;

    @Column(length = 120)
    private String brand;

    @Column(length = 120)
    private String model;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "category_id")
    private ProductCategory category;

    @Column(name = "short_description", length = 500)
    private String shortDescription;

    @Column(columnDefinition = "TEXT")
    private String description;

    @Column(precision = 12, scale = 2)
    private BigDecimal price;

    @Column(nullable = false, length = 8)
    private String currency = "TRY";

    @Column(name = "image_url", length = 500)
    private String imageUrl;

    @Column(nullable = false)
    private Boolean active = true;

    @Column(nullable = false)
    private Boolean featured = false;

    @Column(name = "stock_status", nullable = false, length = 30)
    private String stockStatus = "AVAILABLE";

    @CreationTimestamp
    @Column(name = "created_at", updatable = false, nullable = false)
    private LocalDateTime createdAt;

    @UpdateTimestamp
    @Column(name = "updated_at", nullable = false)
    private LocalDateTime updatedAt;
}

File: frontend/src/types/product.ts
import { PageResponse } from "./api";

/** Backend ProductDto — public katalog */
export interface Product {
  id: number;
  name: string;
  slug: string;
  brand: string | null;
  model: string | null;
  categoryId: number | null;
  categoryName: string | null;
  categorySlug: string | null;
  shortDescription: string | null;
  description: string | null;
  price: number | null;
  currency: string | null;
  imageUrl: string | null;
  active: boolean;
  featured: boolean;
  stockStatus: string | null;
  createdAt: string | null;
  updatedAt: string | null;
}

export type ProductPage = PageResponse<Product>;

export interface ProductCategory {
  id: number;
  name: string;
  slug: string;
  description: string | null;
  sortOrder: number | null;
}

export interface PublicProductFilters {
  category?: string;
  brand?: string;
  search?: string;
  page?: number;
  size?: number;
}
